relay-server/src/services/buffer/envelope_buffer/mod.rs

```diff
@@ -355,14 +361,17 @@ where
     /// Pushes a new [`EnvelopeStack`] with the given [`Envelope`] inserted.
     async fn push_stack(
         &mut self,
+        stack_creation_type: StackCreationType,
```

Comment:
Let's make StackCreationType an enum with data and remove `project_key_pair` and `envelope` parameters, something like

```rust
enum StackOrigin {
    Existing(ProjectKeyPair),
    NewEnvelope(Box<Envelope>),
}
```

Reply:
I think that this is a confusing enumerator since it doesn't depict what is going on. I would rather always create an empty stack and specify the context in which it is created.